### PR TITLE
[13.0][CHG] model name from reason.code to scrap.reason.code

### PR DESCRIPTION
**Pull request by bodedra**
Ref: https://github.com/OCA/stock-logistics-warehouse/pull/770#discussion_r370558559

cc @rousseldenis

### CODE REVIEW

**Comment by bodedra** [comment]
@dreispt @max3903 @Chandresh-SerpentCS are you guys available for a review?

**Comment by AaronHForgeFlow** [comment]
The problem here is current instances with this module installed will need a migration script.

**Comment by bodedra** [comment]
@AaronHForgeFlow Would you please an example?

**Comment by AaronHForgeFlow** [comment]
Sorry I did not explain myself correctly. I actually wanted to say that I am not in favor of such changes in a stable version. Because there is not much benefit. Perhaps the module rename is better to be done in version 14 if the module is migrated. 

Nevertheless, I am not a user of this module so I will not block this PR. You can go ahead with this PR. Just for your information, please notice that you may need migration scripts if you have customers currently using this module, just that.

Regards.

cc @rousseldenis

**Comment by bodedra** [comment]
@AaronHForgeFlow Appreciate your feedback. @rousseldenis What's your call?

**Comment by rousseldenis** [comment]
Indeed, I don't remember that 13.0 version was merged as is by me... IMHO, those modules should be the same. So, could you do migration scripts ?
There is a rename_models method here : https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L149

**Comment by bodedra** [comment]
@rousseldenis Thanks for sharing method. Comment addressed! How about now?

File: scrap_reason_code/migrations/13.0.1.0/pre-migration.py
Comment (rousseldenis) [inline scrap_reason_code/migrations/13.0.1.0/pre-migration.py]: To make this working for existing installs, please change to '13.0.1.1.0' and change it in manifest too.

**Comment by bodedra** [comment]
@rousseldenis @AaronHForgeFlow It seems this is ready for your review. :smile:

**Review by rousseldenis — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Just code review

**Comment by bodedra** [comment]
Ping @dreispt

**Comment by rousseldenis** [comment]
@bodedra Do you keep all those commits ?

**Comment by bodedra** [comment]
@rousseldenis Let's do same thing with this one.

**Review by dreispt — APPROVED** [review]

**Pull request merged**
